src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs

```diff
@@ -109,7 +112,7 @@ private static VSCommanding.CommandState GetCommandState(ITextBuffer buffer)
             return CanExecuteCommand(buffer) ? VSCommanding.CommandState.Available : VSCommanding.CommandState.Unspecified;
         }
 
-        public void ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandler, CancellationToken cancellationToken)
+        public async Task ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandler, CancellationToken cancellationToken)
         {
             // This method handles only return / type char
```

Comment:
❗️ The first line of this method needs to be the following:

```csharp
await _joinableTaskContext.Factory.SwitchToMainThreadAsync();
```

Reply:
reverted